ext/msgpack/factory_class.c

```diff
@@ -113,9 +112,8 @@ VALUE MessagePack_Factory_unpacker(int argc, VALUE* argv, VALUE self)
 
     msgpack_unpacker_t* uk;
     Data_Get_Struct(unpacker, msgpack_unpacker_t, uk);
-
-    msgpack_unpacker_ext_registry_destroy(&uk->ext_registry);
-    msgpack_unpacker_ext_registry_dup(&fc->ukrg, &uk->ext_registry);
+    msgpack_unpacker_ext_registry_borrow(fc->ukrg);
+    uk->ext_registry = fc->ukrg;
```

Comment:
What do you think if `msgpack_unpacker_ext_registry_borrow` will receive `uk->ext_regsitery` pointer and assign `fc->ukrg` into it? I mean L116 should be done in `msgpack_unpacker_ext_registry_borrow` because it should happen always when borrowing `fc->ukrg`.